iconic/UserProfile/views.py

```diff
@@ -173,17 +173,75 @@ def put(self, request):
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
- # Method for deliting city by id  
+    # Method for deliting city by id
     def delete(self, request):
         Cities.objects.get(id=request.data['id']).delete()
         return Response(status=status.HTTP_200_OK)
 
 
-#View for getting all cities
+# View for getting all cities
 class CitiesGet(APIView):
-    
-#Method for getting all cities
+
+    # Method for getting all cities
     def get(self, request):
         cities = Cities.objects.all()
         serializer = CitiesSerializer(cities, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+# A view for getting all users posts
+class AllUserPostsGetView(APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        user_id = request.data['user_id']
+        current_posts = Post.objects.filter(user_id=user_id)
+        data = CustomPostSerializer(current_posts, many=True).data
+        new_data = []
+        for dic in data:
+            dic["likes"] = len(Likes.objects.filter(post_id=dic["id"]))
+            new_data.append(dic)
+        return Response(new_data, status=status.HTTP_200_OK)
+
+
+# A view for getting all information about post
+class PostGetView(APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        post_id = request.data['id']
+        current_post = Post.objects.get(id=post_id)
+        data = CustomPostSerializer(current_post).data
+        data["id"] = post_id
+        data["likes"] = len(Likes.objects.filter(post_id=post_id))
+        return Response(data, status=status.HTTP_200_OK)
+
+
+# a view for changing, adding and deleting data to a post
+class PostEditView(APIView):
+    permissions_classes = (permissions.IsAuthenticated)
+
+    def post(self, request):
+        data = dict(request.data)
+        user_id = request.user.id
+        data["user_id"] = user_id
+        data["likes"] = 0
+        serializer = CustomPostSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(data=serializer.data, status=status.HTTP_201_CREATED)
+        else:
+            return Response(data=serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def put(self, request):
+        instance = Post.objects.get(user_id=request.user.id)
+        serializer = CustomPostSerializer(instance, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(data=serializer.data, status=status.HTTP_202_ACCEPTED)
```

Comment:
In line:
instance = Post.objects.get(user_id=request.user.id)
you should get post by id and request.user.id, not only request.user.id